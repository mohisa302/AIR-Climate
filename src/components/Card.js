import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { useEffect } from 'react';
import { changeTemp } from '../redux/slices/tempSlice';

const Card = (city) => {
  const {
    name,
    temperature,
    icon,
  } = city;

  const dispatch = useDispatch();

  const nameTemp = name.length > 10
    ? `${name.substring(0, 10)}...`
    : name;

  let { europeTemp } = useSelector((state) => state.temp);
  europeTemp = parseInt(europeTemp, 10);
  const data2 = {
    from: '/',
    prop: city,
  };

  useEffect(() => {
    europeTemp += temperature / 60;
    dispatch(changeTemp(europeTemp.toString()));
  }, [dispatch, temperature]);
  return (
    <Link to="/Details" state={data2}>
      <div className="flex flex-col text-right border-2 border-indigo-500/100 p-1">
        <img src={icon} className="h-15 w-20" alt="map" />
        <h6 className="">{nameTemp}</h6>
        <h6>
          {temperature}
          &deg;C
        </h6>
      </div>
    </Link>
  );
};

export default Card;
