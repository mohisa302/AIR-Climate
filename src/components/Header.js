import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Link, NavLink, useLocation } from 'react-router-dom';
import { AiOutlineLeft, AiFillSetting } from 'react-icons/ai';
import { changeCat, serachWord } from '../redux/slices/categorySlice';

const Header = () => {
  const location = useLocation();
  const dispatch = useDispatch();
  const [filter, setFilter] = useState({ category: false, search: false });
  // search input
  const activeLink = location.pathname === '/'
    ? '/'
    : location.pathname.split('').slice(1).join('');

  const geerHandler = (e) => {
    if (e.target.value.toLowerCase() === 'name') {
      dispatch(changeCat({ country: false, all: false, search: '' }));
      setFilter((pre) => ({ search: pre.search, category: !pre.category }));
      console.log(filter);
    } else {
      console.log('l');
      if (e.target.value.toLowerCase() === 'all') dispatch(changeCat({ all: true, country: false, search: '' }));
      if (e.target.value.toLowerCase() === 'country') dispatch(changeCat({ country: true, all: false, search: '' }));
      setFilter((pre) => ({ search: pre.search, category: false }));
    }
  };

  const navLinks = [
    {
      id: 1,
      name: 'Home',
      link: '/',
    },
    {
      id: 2,
      name: 'Details',
      link: '/Details',
    },
  ];

  return (
    <header className="w-full border-b bg-[#210479]">

      <div className="gap-x-4 text-blue-500 sm:flex">
        {navLinks.map((link) => (
          <NavLink
            to={link.link}
            key={link.id}
          />
        ))}
      </div>
      <div className="flex items-center py-3 text-white justify-between gap-x-2 mx-1">
        {activeLink === '/'
            && (
              <h3 className="whitespace-nowrap"> average Temps </h3>
            )}
        {activeLink !== '/'
            && (
            <>
              <Link to="/">
                <AiOutlineLeft />
              </Link>
              <h3> town/city Temp</h3>
            </>
            )}
        <div className="flex items-center">
          {filter.category && (activeLink === '/') && (
          <input
            type="text"
            onChange={(event) => dispatch(serachWord(event.target.value))}
            className="text-black w-32 pl-1 mr-2"
            placeholder="Search..."

          />
          )}
          {filter.search && (activeLink === '/') && (
            <select
              onChange={geerHandler}
              className="text-black"
              data-te-select-init
              data-te-select-placeholder="Select category"
            >
              <option selected>Select</option>
              {['All', 'Country', 'Name'].map((c) => <option className="text-sm" id={`${c}`} value={c} key={c}>{c}</option>)}
            </select>
          )}
          {activeLink === '/' && (
            <AiFillSetting role="tab" className="ml-1 cursor-pointer hover:animate-spin" onClick={() => setFilter((pre) => ({ search: !pre.search, category: false }))} />
          )}
        </div>
      </div>
    </header>
  );
};
export default Header;
