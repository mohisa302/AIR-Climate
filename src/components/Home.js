const Home = () => (
  <div>
    Home
  </div>
);

export default Home;
