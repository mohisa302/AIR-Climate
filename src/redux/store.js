import { configureStore } from '@reduxjs/toolkit';
import weatherReducer from './slices/weatherSlice';

const store = configureStore({
  reducer: {
    climates: weatherReducer,
  },
});

export default store;
