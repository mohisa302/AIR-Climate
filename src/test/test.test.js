/* global expect, jest */
/* global jest */
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import TestRenderer, { act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/Home';
import store from '../redux/store';
import Card from '../components/Card';
import Header from '../components/Header';

describe('Air Climate', () => {
  const climate = [{
    country: 'Germany',
    description: 'clear sky',
    humidity: 88,
    icon: 'https://openweathermap.org/img/wn/01d@2x.png',
    id: 2644210,
    name: 'Liverpool',
    temperature: 4.36,
    wind: 3.6,
  }];

  it('Snapshot of Home Component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot of Header Component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('Navbar card exist in the Header', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    );
    const navElement = screen.getByText(/avera/i);
    expect(navElement).toBeInTheDocument();
  });

  it('Snapshot of Header+Home Component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
          <Header />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Category is all by default on home', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    const category = await screen.getByText(/ALL/i);
    expect(category).toBeInTheDocument();
  });

  it('Gear icon opens selector in the Header', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    );
    act(() => {
      const joinBtns = screen.getAllByRole('tab');
      fireEvent.click(joinBtns[0]);
    });
    const selectElement = document.querySelector('#All');
    expect(selectElement).toBeInTheDocument();
  });

  it('Gear icon closes selector in the Header', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    );
    act(() => {
      const joinBtns = screen.getAllByRole('tab');
      fireEvent.click(joinBtns[0]);
    });

    act(() => {
      const joinBtns = screen.getAllByRole('tab');
      fireEvent.click(joinBtns[0]);
    })
    const selectElement = document.querySelector('#All');
    expect(selectElement).toBeNull();
  });

  it('Snapshot of Card Component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Card name="London" />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
