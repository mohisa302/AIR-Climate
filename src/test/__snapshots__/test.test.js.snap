// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Air Climate Click on Leave button 1`] = `
<a
  href="/Details"
  onClick={[Function]}
>
  <div
    className="flex hover:animate-pulse flex-col text-right object-fill h-full p-1"
  >
    <img
      alt="map"
      className="contrast-100 h-full w-20"
    />
    <h6
      className=""
    >
      Pakistan
    </h6>
    <h6>
      °C
    </h6>
  </div>
</a>
`;

exports[`Air Climate Snapshot of Card Component 1`] = `
<a
  href="/Details"
  onClick={[Function]}
>
  <div
    className="flex hover:animate-pulse flex-col text-right object-fill h-full p-1"
  >
    <img
      alt="map"
      className="contrast-100 h-full w-20"
    />
    <h6
      className=""
    >
      London
    </h6>
    <h6>
      °C
    </h6>
  </div>
</a>
`;

exports[`Air Climate Snapshot of Header Component 1`] = `
<header
  className="w-full border-b bg-[#210479]"
>
  <div
    className="gap-x-4 text-blue-500 sm:flex"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    />
    <a
      className=""
      href="/Details"
      onClick={[Function]}
    />
  </div>
  <div
    className="flex items-center py-3 text-white justify-between gap-x-2 mx-1"
  >
    <h3
      className="whitespace-nowrap"
    >
       average Temps 
    </h3>
    <div
      className="flex items-center"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <svg
          className="mr-4 cursor-pointer hover:animate-spin"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"
          />
        </svg>
      </a>
      <svg
        className="ml-1 cursor-pointer hover:animate-spin"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        role="tab"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512.5 390.6c-29.9 0-57.9 11.6-79.1 32.8-21.1 21.2-32.8 49.2-32.8 79.1 0 29.9 11.7 57.9 32.8 79.1 21.2 21.1 49.2 32.8 79.1 32.8 29.9 0 57.9-11.7 79.1-32.8 21.1-21.2 32.8-49.2 32.8-79.1 0-29.9-11.7-57.9-32.8-79.1a110.96 110.96 0 0 0-79.1-32.8zm412.3 235.5l-65.4-55.9c3.1-19 4.7-38.4 4.7-57.7s-1.6-38.8-4.7-57.7l65.4-55.9a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a442.5 442.5 0 0 0-79.6-137.7l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.2 28.9c-30-24.6-63.4-44-99.6-57.5l-15.7-84.9a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52-9.4-106.8-9.4-158.8 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.3a353.44 353.44 0 0 0-98.9 57.3l-81.8-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a445.93 445.93 0 0 0-79.6 137.7l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.2 56.5c-3.1 18.8-4.6 38-4.6 57 0 19.2 1.5 38.4 4.6 57l-66 56.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.3 44.8 96.8 79.6 137.7l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.8-29.1c29.8 24.5 63 43.9 98.9 57.3l15.8 85.3a32.05 32.05 0 0 0 25.8 25.7l2.7.5a448.27 448.27 0 0 0 158.8 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-84.9c36.2-13.6 69.6-32.9 99.6-57.5l81.2 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.5-87.4 79.6-137.7l.9-2.6c4.3-12.4.6-26.3-9.5-35zm-412.3 52.2c-97.1 0-175.8-78.7-175.8-175.8s78.7-175.8 175.8-175.8 175.8 78.7 175.8 175.8-78.7 175.8-175.8 175.8z"
        />
      </svg>
    </div>
  </div>
</header>
`;

exports[`Air Climate Snapshot of Header+Home Component 1`] = `
Array [
  <main
    className="overflow-auto"
  >
    <div
      className="flex flex-col"
    >
      <div
        className="flex text-blue-900 items-center font-extrabold justify-center gap-x-3"
      >
        <img
          alt="map"
          className="h-40 w-21"
          src="europa.png"
        />
        <div
          className="flex flex-col"
        >
          <h3
            className="font-bold"
          >
            EUROPE
          </h3>
          <h5
            className="text-xs"
          >
            0.00
            °C
          </h5>
        </div>
      </div>
      <div
        className="text-white"
      >
        <h4
          className="bg-[#210479] text-xs font-bold p-1"
        >
          STATS BY 
          <span>
            All
          </span>
        </h4>
        <div
          className="grid grid-cols-2 [&>*:nth-child(4n+3)]:text-[#0a044a] [&>*:nth-child(4n+2)]:text-[#0a044a]  font-semibold md:grid-cols-3 [&>*:nth-child(4n+1)]:bg-[#594acb] [&>*:nth-child(4n)]:bg-[#594acb]"
        />
      </div>
    </div>
  </main>,
  <header
    className="w-full border-b bg-[#210479]"
  >
    <div
      className="gap-x-4 text-blue-500 sm:flex"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      />
      <a
        className=""
        href="/Details"
        onClick={[Function]}
      />
    </div>
    <div
      className="flex items-center py-3 text-white justify-between gap-x-2 mx-1"
    >
      <h3
        className="whitespace-nowrap"
      >
         average Temps 
      </h3>
      <div
        className="flex items-center"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <svg
            className="mr-4 cursor-pointer hover:animate-spin"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"
            />
          </svg>
        </a>
        <svg
          className="ml-1 cursor-pointer hover:animate-spin"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          role="tab"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512.5 390.6c-29.9 0-57.9 11.6-79.1 32.8-21.1 21.2-32.8 49.2-32.8 79.1 0 29.9 11.7 57.9 32.8 79.1 21.2 21.1 49.2 32.8 79.1 32.8 29.9 0 57.9-11.7 79.1-32.8 21.1-21.2 32.8-49.2 32.8-79.1 0-29.9-11.7-57.9-32.8-79.1a110.96 110.96 0 0 0-79.1-32.8zm412.3 235.5l-65.4-55.9c3.1-19 4.7-38.4 4.7-57.7s-1.6-38.8-4.7-57.7l65.4-55.9a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a442.5 442.5 0 0 0-79.6-137.7l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.2 28.9c-30-24.6-63.4-44-99.6-57.5l-15.7-84.9a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52-9.4-106.8-9.4-158.8 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.3a353.44 353.44 0 0 0-98.9 57.3l-81.8-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a445.93 445.93 0 0 0-79.6 137.7l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.2 56.5c-3.1 18.8-4.6 38-4.6 57 0 19.2 1.5 38.4 4.6 57l-66 56.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.3 44.8 96.8 79.6 137.7l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.8-29.1c29.8 24.5 63 43.9 98.9 57.3l15.8 85.3a32.05 32.05 0 0 0 25.8 25.7l2.7.5a448.27 448.27 0 0 0 158.8 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-84.9c36.2-13.6 69.6-32.9 99.6-57.5l81.2 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.5-87.4 79.6-137.7l.9-2.6c4.3-12.4.6-26.3-9.5-35zm-412.3 52.2c-97.1 0-175.8-78.7-175.8-175.8s78.7-175.8 175.8-175.8 175.8 78.7 175.8 175.8-78.7 175.8-175.8 175.8z"
          />
        </svg>
      </div>
    </div>
  </header>,
]
`;

exports[`Air Climate Snapshot of Home Component 1`] = `
<main
  className="overflow-auto"
>
  <div
    className="flex flex-col"
  >
    <div
      className="flex text-blue-900 items-center font-extrabold justify-center gap-x-3"
    >
      <img
        alt="map"
        className="h-40 w-21"
        src="europa.png"
      />
      <div
        className="flex flex-col"
      >
        <h3
          className="font-bold"
        >
          EUROPE
        </h3>
        <h5
          className="text-xs"
        >
          0.00
          °C
        </h5>
      </div>
    </div>
    <div
      className="text-white"
    >
      <h4
        className="bg-[#210479] text-xs font-bold p-1"
      >
        STATS BY 
        <span>
          All
        </span>
      </h4>
      <div
        className="grid grid-cols-2 [&>*:nth-child(4n+3)]:text-[#0a044a] [&>*:nth-child(4n+2)]:text-[#0a044a]  font-semibold md:grid-cols-3 [&>*:nth-child(4n+1)]:bg-[#594acb] [&>*:nth-child(4n)]:bg-[#594acb]"
      />
    </div>
  </div>
</main>
`;
